client/battle/BattleInterfaceClasses.cpp

```diff
@@ -433,6 +433,49 @@ QuickSpellPanel::QuickSpellPanel(BattleInterface & owner)
 	create();
 }
 
+std::vector<std::tuple<SpellID, bool>> QuickSpellPanel::getSpells()
+{
+	std::vector<SpellID> spellIds;
+	std::vector<bool> spellIdsFromSetting;
+	for(int i = 0; i < QUICKSPELL_SLOTS; i++)
+	{
+		std::string spellIdentifier = persistentStorage["quickSpell"][std::to_string(i)].String();
+		SpellID id;
+		try
+		{
+			id = SpellID::decode(spellIdentifier);
+		}
+		catch(const IdentifierResolutionException& e)
+		{
+			id = SpellID::NONE;
+		}	
+		spellIds.push_back(id);	
+		spellIdsFromSetting.push_back(id != SpellID::NONE);	
+	}
+
+	// autofill empty slots with spells if possible
+	auto hero = owner.getBattle()->battleGetMyHero();
+	for(int i = 0; i < QUICKSPELL_SLOTS; i++)
+	{
+		if(spellIds[i] != SpellID::NONE)
+			continue;
+
+		for(auto & availableSpell : CGI->spellh->objects)
+		{
```

Comment:
Please avoid direct access to 'objects' member of our handlers.
This should work, if I haven't missed something:
```suggestion
		for(const auto & availableSpellID : CGI->spellh->getDefaultAllowed())
		{
			const auto * availableSpell = availableSpellID->toSpell();
```